File: anaspoiala-932-dotnet/Lab01_CreationalPatterns/Entities/Coffee.cs
using System.Collections.Generic;
using System.Text;

namespace CoffeeMachine.Entities
{
	public abstract class Coffee
	{
		protected List<CoffeeIngredient> ingredients = new List<CoffeeIngredient>();
		public List<CoffeeIngredient> Ingredients { get => ingredients; }

		public abstract string Drink();

		public override string ToString()
		{
			var builder = new StringBuilder();

			foreach (var ingredient in ingredients)
			{
				builder.AppendLine("- " + ingredient.ToString());
			}

			return builder.ToString();
		}
	}
}

File: src/main/java/ro/ubb/dp1819/lab2/exercises/adapter/src/Coffee/Coffee.java
package Coffee;

public class Coffee {

    private String quantity;
    private String measure;
    private String ingredient;
    private String preference;

    Coffee(String quantity, String measure, String ingredient, String preference) {
        this.quantity = quantity;
        this.measure = measure;
        this.ingredient = ingredient;
        this.preference = preference;
    }

    Coffee(String quantity, String measure, String ingredient) {
        this.quantity = quantity;
        this.measure = measure;
        this.ingredient = ingredient;
    }

    @Override
    public String toString() {
        if(preference!=null)
            return quantity +" "+measure+" coffee with "+ingredient+", "+preference;
        else
            return quantity +" "+measure+" coffee with "+ingredient;
    }
}
